### PR TITLE
Mc question front end

### PR DESCRIPTION
**Pull request by AndrewHall797**
First rendition of the front end for the McQuestion, I tried to emulate the format from the main page

I have a submit button purely for debugging atm, it should be removed when the final page is added later

I also have just made it so that it always posts to a survey with an ID of 1 since I am having issue with the main page, this should of course be replaced later.

I also hardcoded the route, called BaseURL was causing a weird issue, some random text was posted along with the actual url, this should be replaced too.

Currently it can just to the create route.

Screen Shot of rails server creating the entry 

image

Screen Shot of component in browser debugging

image

This was made by looking at tutorials and some stack overflow pages so I am unsure of the quality........

### CODE REVIEW

File: app/javascript/components/CreateSurvey.js [hunk 1]
```diff
@@ -209,6 +210,11 @@ const CreatSurvey = () => {
                                         <DeleteIcon/></Button>
                                     </Stack>
                                     )
+                                case questionType.MULTIPLE_CHOICE:
+                                    return(
+                                        //Need to be replaced with survey id
+                                        <McQuestion survey_id={1}/>
```
Comment (AndrewHall797) [inline app/javascript/components/CreateSurvey.js:216]: this should be replaced

File: app/javascript/components/McQuestion.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+class McQuestion extends React.Component {
+  constructor(props) {
+      super(props);
+      this.state = {question: '', options: [], survey_id: props.survey_id};
+    }
+
+  handleCreateMcQuestion(){
+    //Hardcoded the value for now, Base url causing a weird URL
+		fetch('http://localhost:3000/api/v1/mc_questions/create', {
```
Comment (AndrewHall797) [inline app/javascript/components/McQuestion.js:15]: this should be replaced

File: app/javascript/components/CreateSurvey.js (hunk 1 above)
Comment (AashnaNarang) [reply app/javascript/components/CreateSurvey.js:216]: I believe what should happen is when you first load this page you should call the API to create a new survey with default values (Title = New Title and isLive = false) (this is described in the tech doc btw!). I have very little experience with react and frontend development, but I think that can be done in a componentDidMount() method. Then take the response of that and save the survey id, title, isLive values into the props/state of this file and then on this line, you can make use of the props or state

Comment (AndrewHall797) [reply app/javascript/components/CreateSurvey.js:216]: Yeah, I don't know if the create survey is working properly atm, so this is more of a stub for my own testing.

**Review by JudyHamwi — APPROVED** [review]
Looks good.
one thing we can improve is to have all the returns under one return for the MSQ component.
But it is not obligatory.

**Comment by AndrewHall797** [comment]
>

**Comment by AndrewHall797** [comment]
@JudyHamwi I thought it was all done in one return? Sorry just a bit confused by this.

**Pull request merged**